plugins/org.jboss.tools.openshift.common.core/src/org/jboss/tools/openshift/internal/common/core/util/CommandLocationLookupStrategy.java

```diff
@@ -82,11 +91,32 @@ private  String findLocation(String defaultLoc, String cmd, String which, String
 		}
 		String ret = searchPath(System.getenv(pathvar), delim, cmd);
 		if( ret == null ) {
-			ret = runCommand(which + cmd, timeout);
+			ret = runCommand(which + cmd, timeout, true);
 		}
 		return ret;
 	}
 	
+	public void ensureOnPath( Map<String, String> env, String folder) {
+		String newPath = ensureFolderOnPath(env.get(pathvar), folder);
+		env.put(pathvar, newPath);
+	}
+	
+	public String ensureFolderOnPath(String existingPath, String folder) {
+		existingPath = (existingPath == null ? "" : existingPath);
+		String[] roots = null;
+		if( existingPath.isEmpty() && pathCommand != null) {
+			// Path should never be empty, we have to discover the path (OSX when eclipse launched via .app)
+			String pathresult = runCommand(pathCommand, false);
+			roots = pathresult.split(delim);
+		} else
+			roots = existingPath.split(delim);
```

Comment:
Ok, good.
And well, Javadoc to split method describes its logic in details, it covers all cases that I mentioned above, and that is something worth to remember when it may impact results.